slasher/rpc/update_min_max_span.go

```diff
@@ -0,0 +1,63 @@
+package rpc
+
+import (
+	"github.com/pkg/errors"
+
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+)
+
+// UpdateMaxSpan is used to update the max span of an incoming attestation after the slashing detection phase.
+// logic is following protolambda detection method.
+// from here: https://github.com/protolambda/eth2-surround/blob/master/README.md#min-max-surround
+func (ss *Server) UpdateMaxSpan(source uint64, target uint64, validatorIdx uint64) error {
+	spanMap, err := ss.SlasherDb.ValidatorSpansMap(validatorIdx)
+	if err != nil {
+		return errors.Wrapf(err, "couldnt retrieve span map for validatorIdx: %v", validatorIdx)
+	}
+	if spanMap.EpochSpanMap == nil {
+		spanMap.EpochSpanMap = make(map[uint64]*ethpb.MinMaxSpan)
+	}
+	for i := uint64(1); i < target-source; i++ {
+		val := uint32(target - source - i)
```

Comment:
Why is this cast to a uint32? What if target - source - i is a uint64?

Reply:
to save space in key size. i just need to support 5400 epochs . protobuf doesn't have a uint16. wdyt?